components/skeleton/demo/index.html

```diff
@@ -0,0 +1,112 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+	<meta http-equiv="X-UA-Compatible" content="ie=edge">
+	<meta charset="UTF-8">
+	<link rel="stylesheet" href="../../demo/styles.css" type="text/css">
+	<script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
+	<script type="module">
+		import '../../demo/demo-page.js';
+		import '../../button/button.js';
+		import '../../inputs/input-text.js';
+		import '../../inputs/input-date.js';
+		import './skeleton-host.js';
+	</script>
+	<style>
+		@keyframes loadingPulse {
+			0% { background-color: var(--d2l-color-sylvite); }
+			50% { background: var(--d2l-color-regolith); }
+			75% { background: var(--d2l-color-sylvite); }
+			100% { background: var(--d2l-color-sylvite); }
+		}
+
+		.skeletize::before {
+			content: "";
+			position: absolute;
+			top: 0;
+			bottom: 0;
+			right: 0;
+			left: 0;
+			border-radius: 0.2rem;
+			animation: loadingPulse 1.8s linear infinite;
+			background-color: var(--d2l-color-sylvite);
+			z-index: 2000;
+		}
+
+		.skeletize {
+			position: relative;
+			color: transparent;
+			border: none;
+			box-shadow: none;
+		}
+
+		.flex-container {
+			display: flex;
+			flex-wrap: wrap;
+		}
+
+		.item-container {
+			margin-right: 40px;
+		}
+
+		.item-container[dir="rtl"] {
+			margin-right: 0;
+			margin-left: 40px;
+		}
+
+		d2l-input-text {
+			display: block;
+		}
+
+	</style>
+</head>
+
+<body unresolved>
+	<d2l-demo-page page-title="d2l-skeleton">
+
+		<h2>Custom Element handling skeleton behaviour internally</h2>
+		<d2l-demo-snippet>
+			<template>
+				<d2l-skeleton-host skeleton>
+				</d2l-skeleton-host>
+			</template>
+		</d2l-demo-snippet>
+
+		<h2>Label & Date</h2>
+		<d2l-demo-snippet>
+			<template>
+				<label class="skeletize d2l-label-text">Birth date</label>
+				<d2l-input-date class="skeletize" style="display:block" label="Date" label-hidden></d2l-input-date>
```

Comment:
The duplicate labels here isn't great, and probably won't pass an accessibility audit. We can't have a `<label>` just floating around not associated with a form element, but due to shadow DOM problems we also can't actually link that label with the underlying text input. That's why all the `d2l-input-*` components come with their own label.

But that's OK -- we can just make these components "skeleton aware", and then you could do something like:

```html
<d2l-input-date skeleton style="display:block" label="Birth date"></d2l-input-date>
```